rules/java-net/src/main/resources/otagent/java-net.btm

```diff
@@ -0,0 +1,110 @@
+
+RULE java-net: Java HttpUrlConnection connect entry
+CLASS ^java.net.HttpURLConnection
+METHOD connect()
+HELPER io.opentracing.contrib.agent.OpenTracingHelper
+AT ENTRY
+IF !$0.connected && retrieveSpan($0) == null && !ignore($0)
+DO
+  associateSpan($0, getTracer().buildSpan($0.getRequestMethod())
```

Comment:
it may be worth calling attention (in readme) to the fact that span names will be rather poor with this type of instrumentation. In Jaeger we're currently working on a number of features that have a lot more value when span names are descriptive, but with implicit instrumentation it's always quite hard (we have the same problem with Python monkey-patching of http frameworks).